### DISCUSSION TITLE
Add required method to Request object

### DISCUSSION THREAD

**Discussion by omitobi** [laravel/framework#43058] [Ideas]
Following up on the same proposal on https://github.com/laravel/ideas/issues/2260 

`required` validation rule is the most popular rule that I have seen and this is used often by developers. This feature adds a new `required` method to Request object as a shorthand to "validate" and return required fields.

```php
$request->required(['user_id', 'email']) //returns ['user_id' => 1, 'email' => 'abc@example.com']
```

The above snippet validates both `user_id` and `email `and return an array that is similar to validated() method of `FormRequest` object.

The benefit is that for a little validation that only requires required rule without creating a Form request or writing custom validation.

Proposal for better naming instead of `required` is welcomed.

**Comment by henzeb** [comment]
The use case you and the referenced proposal  give, require additional validation. the user_id must exist, the email must be a valid e-mail. 

Do you have a use case where this is actually useful? I can think of an API-token, but you can already tell if they are there or not using https://laravel.com/docs/master/requests#determining-if-input-is-present

**Reply by omitobi** [reply]
You have a point. I see that $request->filled('name') looks similar to this idea. However, the only thing filled does it to check if the attribute is present and not-empty in the request.

This request->required() would do 4 things

- perform validation for existence
- check if the values are filled
- return the values as array
- throw validation exception if they are invalid

This means we have a shorthand function to both validate, throw validation exception, and retrieved validated (as in with required rule) without adding boilerplate of validation rule.

As for the email and user_id example I gave. Those are just for illustration. The goal is to ensure the required fields are present and filled and we can retrieve them directly.

**Reply by henzeb** [reply]
That's why I asked for a use case. how many use cases can you think of that do not need more validation than just 'required'?

the API-token is a possible use-case, but since laravel has that built in most developers won't use it a lot there.
